File: 2230/2230.js
const fs = require('fs');
const input = fs.readFileSync('input.txt').toString().split('\n');

const [N, M] = input.shift().split(' ').map(Number);
const arr = [];
let answer = Number.MAX_SAFE_INTEGER;

for (let i = 0; i < N; i++) {
  const n = +input.shift();
  arr.push(n);
}

arr.sort((a, b) => {
  return a - b;
});
let [l, r] = [0, 1];

while (r < N) {
  const gap = arr[r] - arr[l];
  if (l == r) r += 1;
  if (gap >= M) {
    answer = Math.min(answer, gap);
    l += 1;
  } else {
    r += 1;
  }
}

console.log(answer);

File: 2230/2230.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;
int N, M;
int answer = 2100000000;
vector<int> arr;

int main()
{
  ios::sync_with_stdio(false);
  cin.tie(0);
  cout.tie(0);
  cin >> N >> M;
  for (int i = 0; i < N; i++)
  {
    int n;
    cin >> n;
    arr.push_back(n);
  }
  sort(arr.begin(), arr.end());

  int l = 0, r = 0;
  while (r < N)
  {
    int gap = arr[r] - arr[l];

    if (gap >= M)
    {
      answer = min(answer, gap);
      if (l == r)
        r += 1;
      else
        l += 1;
    }
    else
    {
      r += 1;
    }
  }
  cout << answer;
}